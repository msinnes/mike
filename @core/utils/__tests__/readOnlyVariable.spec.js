const readOnlyVariable = require('../readOnlyVariable');

describe('readOnlyVariable', () => {
  it('should be a function', () => {
    expect(readOnlyVariable).toBeDefined();
    expect(readOnlyVariable).toBeInstanceOf(Function);
  });

  it('should add the value to the object', () => {
    const obj = {};
    readOnlyVariable(obj, 'prop', 'value');
    expect(obj.prop).toBeDefined();
    expect(obj.prop).toEqual('value');
  });

  it('should throw an error if someone tries to write to the variable', () => {
    const obj = {};
    readOnlyVariable(obj, 'prop', 'value');
    expect(() => {
      obj.prop = 'some new value';
    }).toThrowErrorMatchingSnapshot();
  });

  describe('config', () => {
    describe('get', () => {
      it('should allow for a custom get function', () => {
        let value = 0;
        const obj = {};
        readOnlyVariable(obj, 'prop', value, {
          get: () => value,
        });
        expect(obj.prop).toEqual(0);
        value++;
        expect(obj.prop).toEqual(1);
      });
    });
  });
});