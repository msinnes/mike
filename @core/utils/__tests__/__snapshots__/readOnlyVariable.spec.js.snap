// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readOnlyVariable should throw an error if someone tries to write to the variable 1`] = `"InternalError: prop is read only"`;
